docs/source/guide/events.rst

```diff
@@ -418,7 +455,407 @@ Here is the list of events that users of Boto3 can register handlers to:
+  default settings for retries.
+
+:Keyword Arguments Emitted:
+
+  :type response: tuple
+  :param response: A tuple that includes both the ``botocore.awsrequest.AWSResponse``
+    and a dict that represents the parsed version of the response.
+
+  :type endpoint: ``botocore.endpoint.Endpoint``
+  :param endpoint:  Represents an endpoint for a particular service in a specific
+    region.
+
+  :type operation: ``botocore.model.OperationModel``
+  :param operation: A model representing the underlying API operation of the
+    client method.
+
+  :type attempts: int
+  :param attempts: A number representing the amount of retries that have been attempted.
+
+  :type caught_exception: ``Exception``
+  :param caught_exception: The exception raised after making an api call. If there was no
+    exception, this will be None.
+
+  :type request_dict: dict
+  :param request_dict: A dictionary containing information related to the attempted API request.
+
+:Expected Return Value: Return ``None`` if no retry is needed, or return an ``int`` representing the retry delay.
+
+:Example:
+  Here is an example of how to add a method to a resource class::
+
+    from boto3.session import Session
+
+    def needs_retry_handler(**kwargs):
+        # Implement custom retry logic
+        if some_condition:
+            return None
+        else:
+            return some_delay
+
+    session = Session()
+    session.events.register('needs-retry', needs_retry_handler)
+
+    client = session.client('s3')
+    client.list_buckets()
+
+
+`after-call`
+~~~~~~~~~~~~~~~~~~~~~
+
+:Full Event Name:
+  ``'after-call.service-name.operation-name'``
+
+  .. note::
+
+    ``service-name`` refers to the value used to instantiate a client i.e.
+    ``boto3.client('service-name')``. ``operation-name`` refers to the
+    underlying API operation name of the corresponding client method. 
+
+
+:Description:
+  This event is emitted just after the service client makes an API call.
+  This event allows developers to post process or inspect the API response according to the
+  specific requirements of their application if needed.
+
+:Keyword Arguments Emitted:
+
+  :type http_response: ``botocore.awsrequest.AWSResponse``
+  :param http_response: A data class representing an HTTP response received from the server.
+
+  :type parsed: dict
+  :param params: A parsed version of the AWSResponse in the form of
+    a python dictionary.
+
+  :type model: ``botocore.model.OperationModel``
+  :param model: A model representing the underlying API operation of the
+    client method.
+
+:Expected Return Value: ``None``
+
+:Example:
+  Here is an example that inspects args emitted from the ``after-call`` event::
+
+    from boto3.session import Session
+
+    def print_after_call_args(http_response, parsed, model, **kwargs):
+        print(f'http_response:\n{http_response.text}')
+        print(f'\nparsed:\n{parsed}')
+        print(f'\nmodel:\n{model.name}')
```

Comment:
Separate print statements will already print on a new line so the extra newlines are unnecessary. It might make more sense to keep the key value pairs on the same line. At least for me that's a bit more readable.

Reply:
I simplified the example output format in the latest commit, thanks!